Agent and Backend Ref for Env Vars (#1724)
sensu/sensu-docs: content/sensu-go/5.12/reference/agent.md

  - [Socket configuration](#socket-configuration-flags)
  - [StatsD configuration](#statsd-configuration-flags)
  - [Allow list configuration](#allow-list-configuration)

>   - [Configuring Flags as Environment Variables](#configuring-flags-as-environment-variables)

Reviewer: ```suggestion
  - [Configuration via environment variables](#configuration-via-environment-variables)
```

type                  | Boolean
example               | {{< highlight shell >}}"enable_env": true{{< /highlight >}}

> ### Configuring Flags as Environment Variables

Reviewer: ```suggestion
### Configuration via environment variables
```

> The `sensu-agent` service can read environment variables from `/etc/default/sensu-agent` on Debian/Ubuntu systems and `/etc/sysconfig/sensu-agent` on RHEL systems. These files are not created by the installation package and will need to be created.

Reviewer: ```suggestion
The `sensu-agent` service configured by our supported packages will read environment variables from `/etc/default/sensu-agent` on Debian/Ubuntu systems and `/etc/sysconfig/sensu-agent` on RHEL systems. These files are not created by the installation package and will need to be created.
```

{{< language-toggle >}}

{{< highlight "Ubuntu/Debian" >}}
$ sudo touch /etc/default/sensu-agent
{{< /highlight >}}

{{< highlight "RHEL/CentOS" >}}
$ sudo touch /etc/sysconfig/sensu-agent
{{< /highlight >}}

{{< /language-toggle >}}

> For any flag you are configuring as an environment variable, you will need to append `sensu_` and convert dashes (`-`) to underscores (`_`). This will then need to be added to the appropriate file created above and the service will need to be restarted.

Reviewer: ```suggestion
For any configuration flag you wish to specify as an environment variable, you will need to append `sensu_` and convert any dashes (`-`) to underscores (`_`). The resulting variable will then need to be added to the appropriate environment file described above; the service will need to be restarted before these settings will take effect.
```

In the following example `api-host` flag is configured as an environment variable and set to `"0.0.0.0"`.

{{< language-toggle >}}

{{< highlight "Ubuntu/Debian" >}}

> $ echo 'sensu_api_host="0.0.0.0' | sudo tee /etc/default/sensu-agent

Reviewer: use `tee -a` so that we append to instead of overwrite the existing file

```suggestion
$ echo 'sensu_api_host="0.0.0.0' | sudo tee -a /etc/default/sensu-agent
```

$ sudo systemctl restart sensu-agent
{{< /highlight >}}

{{< highlight "RHEL/CentOS" >}}

> $ echo 'sensu_api_host="0.0.0.0' | sudo tee /etc/sysconfig/sensu-agent

Reviewer: ```suggestion
$ echo 'sensu_api_host="0.0.0.0' | sudo tee -a /etc/sysconfig/sensu-agent
```

$ sudo systemctl restart sensu-agent
{{< /highlight >}}

{{< /language-toggle >}}

> ### Example allow list configuration file

Reviewer: Please move the allow list configuration example above this section so it stays near the allow list configuration specifications. I believe this will also require rearranging the table of contents entries.

{{< /language-toggle >}}

### Configuration via environment variables

The `sensu-agent` service configured by our supported packages will read environment variables from `/etc/default/sensu-agent` on Debian/Ubuntu systems and `/etc/sysconfig/sensu-agent` on RHEL systems. These files are not created by the installation package, so you will need to create them.

{{< language-toggle >}}

{{< highlight "Ubuntu/Debian" >}}
$ sudo touch /etc/default/sensu-agent
{{< /highlight >}}

{{< highlight "RHEL/CentOS" >}}
$ sudo touch /etc/sysconfig/sensu-agent
{{< /highlight >}}

{{< /language-toggle >}}

> For any configuration flag you wish to specify as an environment variable, you will need to append `sensu_` and convert any dashes (`-`) to underscores (`_`). Then, add the resulting environment variable to the appropriate environment file described above. You must restart the service for these settings to take effect.

Reviewer 2: ```suggestion
For any configuration flag you wish to specify as an environment variable, you will need to append `SENSU_` and convert any dashes (`-`) to underscores (`_`). Then, add the resulting environment variable to the appropriate environment file described above. You must restart the service for these settings to take effect.
```

> In the following example, the `API-HOST` flag is configured as an environment variable and set to `"0.0.0.0"`.

Reviewer: ```suggestion
In the following example, the `api-host` flag is configured as an environment variable and set to `"0.0.0.0"`.
```

`api-host` is the parameter name we'd use in agent.yml or as `--api-host` on the command line.
